tests/unit/models/command-test.js

```diff
@@ -558,6 +561,83 @@ describe('models/command.js', function() {
     });
   });
 
+  describe('runTask', function() {
+    let command;
+
+    class AsyncTask extends Task {
+      run(options) {
+        return new Promise(function(resolve) {
+          setTimeout(() => resolve(options), 1000);
+        });
+      }
+    }
+
+    class SyncTask extends Task {
+      run(options) { return options; }
+    }
+
+    class FailingTask extends Task {
+      run(options) { throw new Error('I was born to fail'); }
+    }
+
+    beforeEach(function() {
+      // this should be changed to new Command(), but needs more mocking
+      command = new ServeCommand(Object.assign({}, options, {
+        tasks: {
+          Async: AsyncTask,
+          Sync: SyncTask,
+          Failing: SyncTask,
+        },
+      }));
+    });
+
+    it('always handles task as a promise', function() {
+      command.runTask('Sync', { param: 'value' }).then(result => {
+        expect(result).to.eql({
+          param: 'value',
+        });
+      });
+    });
+
+    it('command environment should be shared with a task', function() {
+      command.runTask('Async', { param: 'value' });
+      expect(command._currentTask.ui).to.eql(command.ui);
+      expect(command._currentTask.analytics).to.eql(command.analytics);
+      expect(command._currentTask.project).to.eql(command.project);
+    });
+
+    it('_currentTask should store a reference to the current task', function() {
+      expect(command._currentTask).to.be.undefined;
+      command.runTask('Sync', { param: 'value' }).then(result => {
+        expect(command._currentTask).to.be.undefined;
+      });
+      expect(command._currentTask).to.be.an.instanceof(SyncTask);
+    });
+
+    it('_currentTask should cleanup current task on fail', function() {
+      command.runTask('Failing', { param: 'value' }).then(result => {
+        expect(command._currentTask).to.be.undefined;
+      });
+    });
+
+    it('throws on attempt to launch concurrent task', function() {
+      try {
```

Comment:
return promises for async tests

I realize the second `runTask` will not create a promise, but the first one will